Cypress v10.11.0

codemechanica opened the pull request:
Hey. Didn't see you at today's meeting. Met up with Josh and we just chatted for 30 minutes. My current status:

- I'm not sure how to write a test to check if an attributes page exists
- I assume that I would need create a new test file for the attributes page in cypress?
- I generated the attributes page
- I created the four input fields and one read only field
- I still can't figure out how to make it reactive by having the value in the input field show up in the read only field
- See you next week Tuesday Nov. 22, 2022

haxwell commented:
Hi Sun.. This looks good so far. Keep filling out those tests!

haxwell commented on src/app/home/home.page.ts:
```diff
@@ -36,9 +36,9 @@ export class HomePage implements OnInit {
-    this._loadingService.show({message: "..loading.."}).then(() => {
-
-    })
+    //this._loadingService.show({message: "..loading.."}).then(() => {
```
This particular change, these three lines, should not go in this PR. Please submit a new PR with these changes.

codemechanica replied:
I am not seeing this

haxwell commented:
Hi Sun.. I think this PR needs to be updated with your recent work in Attributes page.

haxwell commented:
Sun, along with my above comments, there are two *.mp4 video files included in this PR. They should not be. You will need to remove them from the PR, and add an entry to .gitignore to ignore *.mp4 files.

codemechanica commented:
Hi Johnathan. I think there is an issue. I changed remotes so I added a new remote called codemechanica but the the default is set to the old remote which is codice fmigsystems. I added the new remote but I can't seem to change to the new remote. Also, should I ignore ALL mp4 files by adding *.mp4 into the gitignore? I'm not sure why the mp4 files were created by cypress.

haxwell commented:
Hi Sun.. regarding the remote: https://stackoverflow.com/questions/16330404/how-to-remove-remote-origin-from-a-git-repository

Regarding the MP4 files, Cypress creates a video of your E2E test run each time it runs. I believe. and yes, we should ignore all MP4s for now. There may come a time we want to allow them, but not now.

codemechanica commented:
I'm not exactly sure what is wrong. I just did a git remote -v in both the backend and frontend repos and they both show codemechanica as the correct repo. I'll show tomorrow at our meeting.

codemechanica commented on cypress/e2e/attributes.page.cy.ts:
```diff
@@ -0,0 +1,136 @@
+  // check to see if the submit button exists
+  it('check to see if the submit button exists', () => {
+
+    cy.get('[data-test="submitButton"]').should('have.length', 2)
```
say this is from attributes page as a qualifier. I need to name this button uniquely.

codemechanica replied:
shoul have length should only be one

codemechanica replied:
change made

codemechanica commented on cypress/e2e/attributes.cy.ts:
```diff
@@ -0,0 +1,104 @@
+  cy.get('[data-test="inputVerbField"]').type('apple')
+  cy.get('[data-test="inputAdverbButton"]').click()
+  cy.get('[data-test="inputAdverbButton"]').click()
+})
```
I'm not sure if you are looking at my most recent push. I don't have a clear field but I do have a .should() call. Also, I don't have two clicks of the inputAdvertButton. Should I do another push?

codemechanica commented on cypress/e2e/spec.cy.ts:
```diff
@@ -2,4 +2,5 @@ describe('empty spec', () => {
   })
-})
\ No newline at end of file
+})
```
I'm not seeing this on my end. In my code, there is no blank line

codemechanica commented on src/app/home/home.page.html:
```diff
@@ -8,6 +9,8 @@
   This is the home page!
 
+  <ion-button (click)="onGoToAttritubesBtnClick()" data-test="launchAttributesPageButton" fill="clear"><ion-icon name="settings"></ion-icon></ion-button>
+
```
change completed

codemechanica commented on src/app/pages/attributes/attributes.page.html:
```diff
@@ -0,0 +1,47 @@
+      <ion-input id="inputAdverb" data-test="inputAdverbField" (ionInput)="onInputAdverbFieldChange($event)"></ion-input> 
+      <ion-button id="buttonAdverb" data-test="inputAdverbButton" (click)="onInputAdverbButtonClick()">Adverb</ion-button>
+    </ion-item>
+    <!-- 
```
comment removed

codemechanica commented on cypress/e2e/attributes.page.cy.ts:
```diff
@@ -0,0 +1,136 @@
+    cy.get('[data-test="outputField"]').should('have.value', 'banana')
+    
+    
+    // cy.get('[data-test="outputField"]').clear()
```
comment removed

codemechanica commented on cypress/e2e/attributes.page.cy.ts:
```diff
@@ -0,0 +1,136 @@
+
+})
+
+// Must check the change in the input read only field
```
comments removed

codemechanica commented on src/app/pages/attributes/attributes.page.ts:
```diff
@@ -0,0 +1,99 @@
+    return this.outputTxt
+  }
+
+  // making a rest api call
```
removed
